src/mechanics/integrationtypes/IntegrationType1D2NLobatto.cpp

```diff
@@ -0,0 +1,102 @@
+#ifdef ENABLE_SERIALIZATION
+#include <boost/archive/binary_oarchive.hpp>
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/xml_oarchive.hpp>
+#include <boost/archive/xml_iarchive.hpp>
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+#endif //ENABLE_SERIALIZATION
+
+#ifdef ENABLE_VISUALIZE
+#include "visualize/VisualizeEnum.h"
+#endif // ENABLE_VISUALIZE
+
+#include "mechanics/integrationtypes/IntegrationType1D2NLobatto.h"
+#include "mechanics/elements/ElementShapeFunctions.h"
+#include "boost/math/special_functions/legendre.hpp"
+#include "math/Polynomial.h"
+
+// constructor
+NuTo::IntegrationType1D2NLobatto::IntegrationType1D2NLobatto(int numIps)
+{
+    nIps = numIps;
+    std::vector<double> innerNodes = NuTo::Math::Polynomial::legendre_deriv_roots(numIps-1);
+    iPts.push_back(-1.);
+    for (double value : innerNodes) {
+        iPts.push_back(value);
+    }
+    iPts.push_back(1.);
```

Comment:
You can simplify this code a lot by inserting the `(-1)` at the front of the vector.
~~~cpp
nIps = NuTo::Math::Polynomial::legendre_deriv_roots(numIps-1);
nIps.insert(nIps.begin(),-1.);
nIps.push_back(1.);
~~~